kai-client-ci/files.py

```diff
@@ -0,0 +1,96 @@
+import json
+import platform
+import shutil
+import stat
+import sys
+import time
+import zipfile
+from datetime import datetime
+import logging
+import os
+import pandas as pd
+
+import git
+
+import requests
+from dotenv import load_dotenv
+
+from logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def download_file(url, file_path):
+    response = requests.get(url, stream=True)
+    if response.status_code == 200:
+        with open(file_path, 'wb') as file:
+            for chunk in response.iter_content(chunk_size=8192):
+                file.write(chunk)
+        logger.info(f'Downloaded {file_path}')
+    else:
+        logger.error(f'Failed to download the file {file_path} | Status code: {response.status_code}')
+
+
+def rename_extracted_folder(base_folder, target_folder_name):
+    '''
+        Renames the extracted source code folder which is in format konveyor-kai-[uid] to a standard name 'kai'
+    '''
+    extracted_folders = [
+        folder for folder in os.listdir(base_folder)
+        if os.path.isdir(os.path.join(base_folder, folder)) and folder.startswith('konveyor-kai-')
+    ]
+
+    original_folder_path = os.path.join(base_folder, extracted_folders[0])
+    target_folder_path = os.path.join(base_folder, target_folder_name)
+
+    try:
+        shutil.move(original_folder_path, target_folder_path)
+    except Exception as e:
+        logger.error(f'Failed to rename folder: {e}')
+
+
+def clone_repository(app_name, repository_url, branch):
+    clone_dir = os.path.join('data', f'{app_name}')
+    git.Repo.clone_from(repository_url, clone_dir, branch=branch)
+    logger.info(f"Repository {app_name} cloned into 'data' with branch 'main'")
+
+
+def unzip_file(zip_path, extract_folder):
+    try:
+        with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+            zip_ref.extractall(extract_folder)
+        logger.info(f'Extracted {zip_path} into {extract_folder}')
+    except zipfile.BadZipFile:
+        logger.error(f'Failed to extract {zip_path}')
+
+
+def zip_folder(file_path, file_name):
+    zip_filename = f'{file_name}.zip'
+    zip_path = os.path.join('data', zip_filename)
+
+    shutil.make_archive(zip_path.replace('.zip', ''), 'zip', file_path)
```

Comment:
Does this overwrite existing files with same name ? Is there possibility for files with same name ? If yes then we need to remove existing files before writing

Reply:
It will throw an exception, right now we use it just for zipping the results and we are naming them with the timestamp so there won't be duplicates.